src/resources/timeSeries/timeSeriesApi.ts

```diff
@@ -87,37 +90,81 @@ export class TimeSeriesAPI {
    */
   public delete: TimeSeriesDeleteEndpoint;
 
+  private client: CogniteClient;
+
   /** @hidden */
-  constructor(project: string, instance: AxiosInstance, map: MetadataMap) {
+  constructor(
+    client: CogniteClient,
+    project: string,
+    instance: AxiosInstance,
+    map: MetadataMap
+  ) {
+    this.client = client;
     const path = projectUrl(project) + '/timeseries';
-    this.create = generateCreateEndpoint(instance, path, map);
-    this.list = generateListEndpoint(instance, path, map, false);
-    this.search = generateSearchEndpoint(instance, path, map);
-    this.retrieve = generateRetrieveEndpoint(instance, path, map);
-    this.update = generateUpdateEndpoint(instance, path, map);
+    const transformResponse = this.transformToTimeSeriesListObject;
+    this.create = generateCreateEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse
+    );
+    this.list = generateListEndpoint(
+      instance,
+      path,
+      map,
+      false,
+      transformResponse
+    );
+    this.search = generateSearchEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse
+    );
+    this.retrieve = generateRetrieveEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse
+    );
+    this.update = generateUpdateEndpoint(
+      instance,
```

Comment:
all (except `generateListEndpoint`) `generate...()` calls pass the same parameters `instance, path, map, transformResponse`
could we maybe add them to an array and pass that `...array`
like it's done here
```
    const defaultArgs: [string, AxiosInstance, MetadataMap] = [
      this.project,
      this.instance,
      this.metadataMap,
    ];
    this.assetsApi = new AssetsAPI(this, ...defaultArgs);
```

Reply:
and then for `generateListEndpoint ` you can keep as it is

Reply:
Done

Reply:
Should the same thing be done for `AssetApi` tho?